tests/test_infer.py

```diff
@@ -85,3 +85,61 @@ def test_hypotest_return_expected_set(tmpdir, hypotest_args):
     assert isinstance(result[2], type(tb.astensor(result[2])))
     assert len(result[3]) == 5
     assert check_uniform_type(result[3])
+
+
+def test_inferapi_pyhf_independence():
+    '''
+    pyhf.infer should eventually be factored out so it should be
+    infependent from pyhf internals. This is testing that
+    a much simpler model still can run through pyhf.infer.hypotest
+    '''
+    from pyhf import get_backend
+
+    class _NonPyhfConfig(object):
+        def __init__(self):
+            self.poi_index = 0
+            self.npars = 2
+
+        def suggested_init(self):
+            return [1.0, 1.0]
+
+        def suggested_bounds(self):
+            return [[0.0, 10.0], [0.0, 10.0]]
+
+    class NonPyhfModel(object):
+        def __init__(self, spec):
+            self.sig, self.nominal, self.uncert = spec
+            self.factor = (self.nominal / self.uncert) ** 2
+            self.aux = 1.0 * self.factor
+            self.config = _NonPyhfConfig()
+
+        def _make_main_pdf(self, pars):
+            mu, gamma = pars
+            expected_main = gamma * self.nominal + mu * self.sig
+            return pyhf.probability.Poisson(expected_main)
+
+        def _make_constraint_pdf(self, pars):
+            mu, gamma = pars
+            return pyhf.probability.Poisson(gamma * self.factor)
+
+        def expected_data(self, pars, include_auxdata=True):
+            tensorlib, _ = get_backend()
+            expected_main = tensorlib.astensor(
+                [self._make_main_pdf(pars).expected_data()]
+            )
+            aux_data = tensorlib.astensor(
+                [self._make_constraint_pdf(pars).expected_data()]
+            )
+            if not include_auxdata:
+                return expected_main
+            return tensorlib.concatenate([expected_main, aux_data])
+
+        def logpdf(self, pars, data):
+            tensorlib, _ = get_backend()
+            maindata, auxdata = data
+            main = self._make_main_pdf(pars).log_prob(maindata)
+            constraint = self._make_constraint_pdf(pars).log_prob(auxdata)
+            return tensorlib.astensor([main + constraint])
+
+    m = NonPyhfModel([5, 50, 7])
```

Comment:
```suggestion
    model = NonPyhfModel([5, 50, 7])
```

Prefer to not have single letter variables. :)